connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java

```diff
@@ -50,17 +52,40 @@ public InboundCorrelationHandler(ZeebeClient zeebeClient, FeelEngineWrapper feel
     this.feelEngine = feelEngine;
   }
 
-  public CorrelationResult correlate(InboundConnectorDefinitionImpl definition, Object variables) {
-    return correlate(definition, variables, null);
+  public CorrelationResult correlate(InboundConnectorDefinitionImpl definitions, Object variables) {
+    return correlate(definitions, variables, null);
   }
 
   public CorrelationResult correlate(
-      InboundConnectorDefinitionImpl definition, Object variables, String messageId) {
+      InboundConnectorDefinitionImpl definitions, Object variables, String messageId) {
 
-    var correlationPoint = definition.correlationPoint();
+    List<InboundConnectorElement> matchingElements =
+        definitions.elements().stream()
+            .filter(e -> isActivationConditionMet(e, variables))
+            .map(e -> (InboundConnectorElement) e)
+            .toList();
+
+    if (matchingElements.isEmpty()) {
+      return ActivationConditionNotMet.INSTANCE;
+    }
+    if (matchingElements.size() > 1) {
+      return new Failure.InvalidInput("Multiple connectors are activated for the same input", null);
+    }
+
+    var activatedElement = matchingElements.getFirst();
+    return correlateInternal(
+        matchingElements, (InboundConnectorElementImpl) activatedElement, variables, messageId);
```

Comment:
Curious in general whether we benefit from this interface here (and other impl as well) as seeing this casting usually implies we dont.

Reply:
True, I'll check how we can improve this and get rid of casts. The main purpose it to strictly limit what we expose to the connector user space vs. what is available to the runtime

Reply:
Sure. Maybe we can introduce interfaces when exposing information via a context but use basic records to store immutable data internally?